### PR TITLE
Fix: M2-2335 - Dashboard - Changes in Applet table to support backend fixes for folders, search, and pagination

### PR DESCRIPTION
**Pull request by muhurov**

### CODE REVIEW

File: src/modules/Dashboard/features/Applets/Applets.tsx [hunk 1]
```diff
@@ -62,9 +60,9 @@ export const Applets = () => {
   const addFolder = () => {
     const newFolderName = generateNewFolderName(folders, t);
     const folderRow = {
-      appletCount: 0,
+      foldersAppletCount: 0,
       id: (Math.random() + Math.random()).toString(),
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/Applets/Applets.tsx:64]: maybe change to uniqueId?

Comment (muhurov) [reply src/modules/Dashboard/features/Applets/Applets.tsx]: done

File: src/modules/Dashboard/features/Applets/Applets.hooks.ts [hunk 2]
```diff
@@ -6,56 +6,84 @@ import {
...
+    const getAppletsList = search ? getFilteredWorkspaceApplets : getWorkspaceApplets;
+
     const {
       data: { result: folders },
     } = await getWorkspaceFolders({ ownerId: ownerId! });
     const {
-      data: { result: applets },
-    } = await getWorkspaceApplets(
+      data: { result: applets, count },
+    } = await getAppletsList(
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/Applets/Applets.hooks.ts:43]: why not to use `Promise.all(getWorkspaceFolders ... , getAppletsList ... )` ?

File: src/modules/Dashboard/features/Applets/Applets.hooks.ts [hunk 3]
```diff
@@ -6,56 +6,84 @@ import {
...
-      ...applets.map((applet) => ({ ...applet, isFolder: false })),
-    ];
+    setCount(count);
 
-    if (!expandedFolders.length) {
-      setRows(formattedApplets);
+    const appletList =
+      applets?.map((applet) => ({
+        ...applet,
+        isFolder: applet.type === 'folder',
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/Applets/Applets.hooks.ts:54]: if it is applicable, let's create a enum for applet.type?

File: src/modules/Dashboard/features/Applets/Applets.hooks.ts (hunk 2 above)
Comment (muhurov) [reply src/modules/Dashboard/features/Applets/Applets.hooks.ts]: done

File: src/modules/Dashboard/features/Applets/Applets.hooks.ts (hunk 3 above)
Comment (muhurov) [reply src/modules/Dashboard/features/Applets/Applets.hooks.ts]: fixed

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**